File: runme/mcgen/local/datan/exasrc/e1lsq.c
#include <stdio.h>
#include <math.h>
#include "datsrc.h"

/* Table of constant values */

static integer c__10 = 10;

main(void)
{
    /* Initialized data */ /* Local variables */
    static integer i, k, nr;
    static integer c__1 = 1;
    static double t[10] = { -.9,-.7,-.5,-.3,-.1,.1,.3,.5,.7,.9 },
                  y[10] = { 81.,50.,35.,27.,26.,60.,106.,189.,318.,520. },
                  a[40] /* was [10][4] */, scrat[16] /* was [4][4] */,
                  cx[16] /* was [4][4] */, r, x[4], deltay[10];
    static logical ok;

/* identify program to user */
    printf("%s\n\n", "Program E1LSQ demonstrates use of LSQPOL.");
/* list data */
    printf("%s\n", "   T  ,      Y       DELTAY");
    for (i = 1; i <= 10; ++i) {
        deltay[i - 1] = sqrt(y[i - 1]);
        printf("%10.5lf%10.5lf%10.5lf\n", t[i-1], y[i-1], deltay[i-1]);
    }
    printf("\n");
/* perform fits with polynomials of different degree */
    for (nr = 1; nr <= 4; ++nr) {
        lsqpol_(t, y, deltay, &c__10, &nr, x, cx, &r, a, scrat, &ok);
        printf("%s%1li%s%10.5lf\n", "Polynomial fit with NR = ", nr,
               ", M = ", r);
        printf("%s\n", "X = ");
        mtxwrt_(x, &c__1, &nr);
        printf("\n");
        printf("%s\n", "Covariance matrix CX =");
        mtxwrt_(cx, &nr, &nr);
        printf("\n");
    }
    return 0;
} /* main */

File: runme/mcgen/local/datan/exasrc/e1lsq.f
      PROGRAM E1LSQ
      IMPLICIT DOUBLE PRECISION(A-H,O-Z)
      PARAMETER(MAXNR=4,MAXN=10,N=10)
      DIMENSION T(MAXN),DELTAY(MAXN),CX(MAXNR,MAXNR),A(MAXN,MAXNR)
      DIMENSION Y(MAXN),X(MAXNR),SCRAT(MAXNR,MAXNR)
      LOGICAL OK
      DATA T /-.9D0,-.7D0,-.5D0,-.3D0,-.1D0,
     +.1D0,.3D0,.5D0,.7D0,.9D0/
      DATA Y /81.D0,50.D0,35.D0,27.D0,26.D0,
     +60.D0,106.D0,189.D0,318.D0,520.D0/
C identify program to user
      WRITE(*,*)' Program E1LSQ demonstrates use of LSQPOL'
      WRITE(*,*)' '
C list data
      WRITE(*,*)'   T  ,      Y       DELTAY'
      DO 10 I=1,MAXN
          DELTAY(I)=SQRT(Y(I))
        WRITE(*,'(3F10.5)')T(I),Y(I),DELTAY(I)
10    CONTINUE
      WRITE(*,*)' '
C perform fits with polynomials of different degree
      DO 20 NR=1,MAXNR
        CALL LSQPOL(T,Y,DELTAY,N,NR,X,CX,R,A,SCRAT,OK)
        WRITE(*,'(A,I1,A,F10.5)')' Polynomial fit with NR = ',NR,
     +  ', M = ',R
        WRITE(*,'(A))')' X = '
        CALL MTXWRT(X,1,NR)
        WRITE(*,*)' Covariance matrix CX ='
        CALL MTXWRT(CX,NR,NR)
        WRITE(*,*)' '
20    CONTINUE
      END
